app/internal/page/certificateprovider/register.go

```diff
@@ -180,8 +191,13 @@ func makeHandle(mux *http.ServeMux, store sesh.Store, errorHandler page.ErrorHan
 
 				appData.SessionID = session.DonorSessionID
```

Comment:
This is the sin we want to remove. `SessionID` for certificate providers should be based on their `sub`, we shouldn't be passing the donor's around to link as we have the `LpaID` for that.